libraries/AP_BattMonitor/AP_BattMonitor_SMBus.cpp

```diff
@@ -22,6 +25,25 @@ void AP_BattMonitor_SMBus::read(void)
     if (_serial_number && _mon._serial_numbers[_state.instance] == -1) {
         _mon._serial_numbers[_state.instance].set_and_notify(_serial_number);
     }
+
+    if (_full_charge_capacity != _mon._pack_capacity[_state.instance]) {
```

Comment:
I'd prefer it did the same check as above: check if the parameter has the default value and whether we got the full charge capacity.